tests/acceptance/01_vars/02_functions/variablesmatching.cf

```diff
@@ -21,25 +25,15 @@ bundle agent test
       "x_vars" slist => variablesmatching("default:init.test_fbeae67f3e347b5e0032302200141131.*", "x");
       "z_vars" slist => variablesmatching("default:init.test_fbeae67f3e347b5e0032302200141131.*", "z");
 
-      "count" int => length(vars);
-      "x_count" int => length(x_vars);
-      "z_count" int => length(z_vars);
+      "fullvars" data => data_matching("default:init.test_b37ce1b03955522848f0a82d0b2b2e21.*");
```

Comment:
1. I find this confusing. What I was expecting from a function named `data_matching()` was matching the argument with the variables' contents. Instead what I see is that it matches exactly what `variablesmatching()` matches, but returns a data container instead of an slist.

2. Why is this useful? This does not address the main concern of the ticket it mentions (CFE-2265, old Redmine#7885) which is to avoid double-dereferencing the variables; you'll still need to do `getvalues()`on the returned data container.